Chapter I: Introduction (#8)
LeidongX/Leidong_Thesis: tex/chapters/chapter1.tex

\begin{equation}
 \mathbf{(I - DL^{-1}MS)} \mathbf{\mathbf{x}} = \frac{1}{k} \mathbf{DL^{-1}MF} \mathbf{\mathbf{x}}  \, ,
 \label{eq:keig}
\end{equation}
where $\mathbf{\mathbf{x}} $ describes the scalar flux, $\mathbf{L}$ represents loss operator,  $\mathbf{F}$ represents fission operator, $\mathbf{S}$ represents the scattering operator, $\mathbf{M}$ and $\mathbf{D}$ represent moment-to-discrete and discrete-to-moment operators respectively, and the eigenvalue $k$ represents the ratio of gains to losses.
A reactor is ``critical,'' i.e., the neutron population is steady, when $k = 1$; or ``subcritical,'' i.e., the neutron density is decreasing as time passes when $k < 1$; or ``supercritical,'' i.e., the neutron density is increasing with time when $k>1$.
The eigenvector corresponding to the largest eigenvalue $k$, which is usually known as fundamental (or dominant) eigenmode, indicates the scalar flux distribution when the system reaches steady state.
A more generalized multigroup neutron transport equation will be presented in Chapter 2.

\section{Motivation and Objective}
The discrete ordinates method is a practical, deterministic approach that is used to solve \EQ{eq:keig}.

> This approach is often smoothed by other iterative methods such as the power method (PM), Richard iteration method, or the generalized minimal residual method (GMRES)\cite{saad1986gmres}, which each apply the loss operator $L$ every iteration.

Author: I add a citation for Richarson, but the Power method is too well-known and I did not see anyone cite where it originally invented.

> This approach is often smoothed by other iterative methods such as the power method (PM), Richard iteration method, or the generalized minimal residual method (GMRES)\cite{saad1986gmres}, which each apply the loss operator $L$ every iteration.

Reviewer: You could always cite a math text book.  I'm still not sure what you mean by `smoothed`.

Numerically, it is inefficient to solve for the operator inverse; instead, a series of sweeps over each energy group and space location can be performed until the system converges.  
However, the expense of this approach is now proportional to the number of iterations.
Many breakthrough, such as the generalized Davidson(GD) method\cite{larsen1984diffusion} and Diffusion Synthetic Acceleration(DSA)\cite{hamilton2011numerical}, came to accelerate the aforementioned iteration methods.

Within the development of computational science，some theory of data science has been employed to analyze engineering problems.

> Dynamic mode decomposition(DMD) is one of the most well-known supervised learning approaches that has achieved great success in computational fluid dynamics.

Reviewer: ```suggestion
Dynamic mode decomposition(DMD) is one of the most well-known, supervised learning approaches that has achieved great success in computational fluid dynamics.
```

The primary focus of this thesis is to estimate accurate fundamental eigenmodes by DMD to accelerate the Power Method and other, related methods.

To achieve these goals, a review of DMD will be described in the remainder of this chapter.
\CHAPTER{chapter:multigroup} will introduce the multigroup neutron transport and diffusion equations, and \CHAPTER{chapter:PM} discusses the mathematical background of the power and flatten power methods.
Following that, a restarted, DMD-accelerated power method scheme (DMD-PM(n)) is presented in \CHAPTER{chapter:DMD-PM}, while a DMD-based, accelerated, flattened power
method DMD-FPM(n) is presented in \CHAPTER{chapter:DMD-FPM(n)}.
Then, the results using either of acceleration schemes on compute analyzing a 1-D boiling water reactor(BWR) model and the famous 2-D C5G7 test problem, are discussed in \CHAPTER{chapter:results}.
\CHAPTER{chapter:conclusion} will include conclusions and future work including the possibility of combining this theory with other methods. 

\section{Summary of previous work}
Before continuing, it would be useful to briefly introduce some great achievements that demonstrated the success of dynamic mode decomposition.

> DMD was initially proposed by \citet{schmid_dynamic_2010}\cite{schmid_applications_2011} to extract dynamics information from time-series data of fluids observations in 2010.

Reviewer: I think you want this to be `\citet{schmid_dynamic_2010, schmid_applications_2011}`, but look at the compiled version to be sure

Author: this way will show the name twice, so I just keep it so far.

He applied this frame on both numerical Navier-Stokes code results and experimental measures and illustrated the significant essential on quantifying fluid-dynamical problems.
As opposed to proper orthogonal decomposition (POD)\cite{lumley2007stochastic}, DMD is a pure data-based procedure and does not decompose on an orthogonal space. 

> This method is based on Koopman analysis of nonlinear dynamical systems and can extracte dynamic modes to describe the global behavior captured by projecting large-scale problems onto a reduced order dynamical system.

Reviewer: I've no idea what Koopman analysis means, so you may wish to add a bit more detail here.

Author: It's a very complicated theory and cannot be summarized in short, so I add a citation here

In his work, the time-dependent dynamic systems are described in the form of a snapshots sequence,
\begin{equation}
 \mathbf{X}^{N}_1 = \{\mathbf{x}_0, \mathbf{x}_1, \ldots, \mathbf{x}_{N} \} \, ,
 \label{eq:snap_matrix}
\end{equation}
where $\mathbf{x}_i$ stands for the {\it i}th state.
We assume that there exist a linear mapping operator $\mathbf{A}$, which can be applied on on the initial data snapshot $\mathbf{x}_0$ repeatedly and formulate snapshots sequence \EQ{eq:snap_matrix} as a Krylov sequence by
\begin{equation}
 \mathbf{X}^{N}_1 = \{\mathbf{x}_1,A\mathbf{x}_1,A^2\mathbf{x}_1,…,A^{N-1}\mathbf{x}_1 \} \, .
 \label{eq:Krylov_seq}
\end{equation}

> Some famous older attempts, like Arnoldi method\cite{arnoldi1951principle}, uses QR-decomposition $\mathbf{QR} = \mathbf{X}^{N-1}_1$ to reconstruct the mapping operator, which is not applicable while the matrix $A$ has to be available explicitly for the projections.\cite{Greenbaum_1997}  \cite{trefethen1997numerical}

Reviewer: The last part of this sentence is not correct grammatically, but I can't fix it because I'm unsure of the meaning (I can see multiple possible corrections).  What does the part after `the mapping operator` mean?

Author: edited

Instead of employing the QR-decomposition, Schmid took advantage of the SVD decomposition and obtained a robust approximation by 
\begin{equation}
\mathbf{\tilde{A}} = \mathbf{U^H X_1^{N-1}V\Sigma^{-1}} \, ,
 \label{eq:stanard_DMD}
\end{equation}
where $\mathbf{X}_1^{N-1} = \mathbf{U}\Sigma \mathbf{V}^H$. 
The eigenvector $x_i$ of $\mathbf{\tilde{A}}$ and left eigenvector matrix $\mathbf{U}$ are used to calculate the DMD modes $\mathbf{\mathbf{x}}_i$ by 
\begin{equation}
\mathbf{\mathbf{x}}_i = \mathbf{U} \mathbf{x}_i \, ,
 \label{eq:dyanmic_modes}
\end{equation}

> and, therefore, recover the a reduced order dynamic system.

Reviewer: ```suggestion
and, therefore, recover the reduced order dynamic system.
```

This scheme is often known as the standard DMD approach, and a more detailed discussion of the standard DMD scheme is presented in \CHAPTER{chapter:DMD-PM}.   

> Many practical theory were developed based on the this successfully attempt, and past and future snapshot matrix is defined as $\mathbf{X}_0 \triangleq \{\mathbf{x}_0, \mathbf{x}_1, \ldots, \mathbf{x}_{N-1} \}$ and $\mathbf{X}_1 \triangleq \{\mathbf{x}_1, \mathbf{x}_1, \ldots, \mathbf{x}_{N} \} $. 

Reviewer: ```suggestion
Many practical theories were developed based on the this successfully attempt.
One such attempt uses the past and future snapshot matrix, which is defined as $\mathbf{X}_0 \triangleq \{\mathbf{x}_0, \mathbf{x}_1, \ldots, \mathbf{x}_{N-1} \}$ and $\mathbf{X}_1 \triangleq \{\mathbf{x}_1, \mathbf{x}_1, \ldots, \mathbf{x}_{N} \} $. 
```

Reviewer: The second sentence doesn't seem to make sense in this location, however.

> \citet{tu_dynamic_2014} posted a delicate mathematical analysis of the DMD algorithm and develop a theoretical extension of DMD(extract DMD) using a new definition of operater $\mathbf{A}$ by    

Reviewer: ```suggestion
\citet{tu_dynamic_2014} posted a delicate mathematical analysis of the DMD algorithm and developed a theoretical extension of DMD(extract DMD) using a new definition of operater $\mathbf{A}$ by    
```

\begin{equation}
 \mathbf{A}\triangleq \mathbf{X}_1  \mathbf{X}_0^{\dagger}\, ,
 \label{eq:exact_dmd}
\end{equation}

> where $\mathbf{X}_0^{\dagger}$ is the pseudoinverse of $\mathbf{X}_0$. And the DMD modes and

Reviewer: ```suggestion
where $\mathbf{X}_0^{\dagger}$ is the pseudoinverse of $\mathbf{X}_0$. 
The DMD modes and
```

> eigenvalues are the eigenvectors and eigenvalues of $\mathbf{A}$, which can be computed by eigendecomposition. However, in practice, it may be expensive to compute the eigendecomposition of $\mathbf{A}$. With very minimum changes, exact DMD can also compute the modes and eigenvalue without manipulate explicit $\mathbf{A}$. In the case of directly-manipulation-free extract DMD, all other processes keep consistent with standard DMD, but the DMD modes are computed by 

Reviewer: ```suggestion
eigenvalues are the same as the eigenvectors and eigenvalues of $\mathbf{A}$, which can be computed by eigendecomposition.
However, in practice, it may be expensive to compute the eigendecomposition of $\mathbf{A}$.
With very minimum changes, exact DMD can also compute the modes and eigenvalue without explicitly forming $\mathbf{A}$.
In the case of directly-manipulation-free extract DMD, all processes are consistent with standard DMD, except for computation of the DMD modes, which is now done by
```

\begin{equation}
 \mathbf{\mathbf{x}}_i = \frac{1}{\lambda} \mathbf{X}_1 \mathbf{V} \mathbf{\Sigma}^{-1} \omega \, .
 \label{eq:exact_dmd_free}
\end{equation}

\citet{kutz_dynamic_2016} summarized many other fashion variety DMD algorithms and discussed the applicability fitting with different complex systems in a recent monograph.
Fundamental theoretical foundations of DMD and the Koopman operator was also developed in this book.

\subsection{DMD in Nuclear Engineering}

> Soon after, this technology then was applied in nuclear reactor simulation. 

Reviewer: ```suggestion
Soon after, this technology then was applied to nuclear reactor simulations. 
```

> Abdo and others used DMD as a direct, explicit-in-time surrogate for such black-box models, e.g., to model the evolution of nuclear reactor isotopics over long time periods; and the nonlinear response of reactor power during short transients.\cite{abdo_data-driven_2018}\cite{abdo_modeling_2019}\cite{elzohery2018comparison}

Reviewer: ```suggestion
\citet{abdo_data-driven_2018} used DMD as a direct, explicit-in-time surrogate for black-box models, e.g., to model the evolution of nuclear reactor isotopics over long time periods as well as the nonlinear response of reactor power during short transients.\cite{abdo_modeling_2019}\cite{elzohery2018comparison}
```

\subsection{DMD Acceleration}

> A way to achieve acceleration is to use results from lower dimensional system projected system to find a correction each iteration until system converges.

Reviewer: Acceleration of what?  Be more specific.

Author: good point. Edited

The first research of using DMD to accelerate iterative methods was the work of \citet{andersson_novel}.

> A Krylov subspace is built up by store the states samples with time sequence as $\mathbf{V_n}$, then Gram-Schmidt algorithm is applied by $\mathbf{V_n} = \mathbf{{E_n U_n}}$.

Reviewer: You may wish to explain a bit more about what a Krylov subspace is.

Author: ref to an equation discussed earlier.

> A Krylov subspace is built up by store the states samples with time sequence as $\mathbf{V_n}$, then Gram-Schmidt algorithm is applied by $\mathbf{V_n} = \mathbf{{E_n U_n}}$.

Reviewer: ```suggestion
A Krylov subspace is built up by storing the state-samples with time sequence as $\mathbf{V_n}$.
Then the Gram-Schmidt algorithm is applied by $\mathbf{V_n} = \mathbf{{E_n U_n}}$ to orthogonalize the subspace.
```

> The reason why authors would not like to use SVD is that Gram-Schmidt algorithm only work with two state vectors each time and SVD requires the whole subspace matrix $V_n$. 

Reviewer: ```suggestion
Most applications avoid the SVD because the Gram-Schmidt algorithm only works with two state vectors each time, and the SVD requires the whole subspace matrix $V_n$. 
In other words, Gram-Schmidt is slow for large systems.
```

> Moreover, Gram-Schmidt algorithm is obviously a more suitable approach for parallel computations. 

Reviewer: I got the impression from the previous sentence that Gram-Schmidt is bad, but this sentence says otherwise.

Author: deleted

Reviewer: I don't think it was deleted in this version.  Did you forget to push a commit?

Author: The sentence that said Gram-Schmidt is slow is wrong. I deleted that one.

The reduced projection of the system matrix $\mathbf{\tilde{A}}$ is obtained by 
\begin{equation}
 \mathbf{\tilde{A}} = \mathbf{E}_n^T \mathbf{V}_{n+1} \mathbf{U}_n^{-1}  \, ,
 \label{eq:andersson_reduce}
\end{equation}
and is implemented into iterative linear relation $\mathbf{x}^{(n+1)} = \mathbf{A}\mathbf{x}^{(n)} + \mathbf{b}$.
The estimate of the converged solution can be calculate in the modified form as:
\begin{equation}

>  \mathbf{x}^{u} = \mathbf{x}^{(n+1)} + \mathbf{E}_n(\mathbf{I_n} - \mathbf{\tilde{A}}_n)^{-1} \mathbf{E}_n^T(\mathbf{x}^{(n+2)} - \mathbf{x}^{(n+1)}) \, ,

Reviewer: ```suggestion
 \mathbf{x}^{u} = \mathbf{x}^{(n+1)} + \mathbf{E}_n(\mathbf{I_n} - \mathbf{\tilde{A}}_n)^{-1} \mathbf{E}_n^T(\mathbf{x}^{(n+2)} - \mathbf{x}^{(n+1)}) \, .
```

 \label{eq:andersson}
\end{equation}

> By performing this process several time to correct the solution, both the number of iterations and the average fluctuation was reduced almost 30\% for compressible flow problems. 

Reviewer: ```suggestion
By performing this process several times to correct the solution, both the number of iterations and the average fluctuation were reduced almost 30\% for compressible flow problems. 
```

> Then \citet{mcclarren_acceleration_2018} presented a novel acceleration technique for improving the convergence of Richardson(source) iteration for source-driven neutronics problems.

Reviewer: ```suggestion
Then, \citet{mcclarren_acceleration_2018} presented a novel acceleration technique for improving the convergence of Richardson(source) iteration for source-driven neutronics problems.
```

The Richardson iteration can be expressed as 
\begin{equation}
 \mathbf{x}^{(n+1)} = (\mathbf{I}-\mathbf{A})\mathbf{x}^{(n)} + \mathbf{b} \, ,
 \label{eq:richardson}
\end{equation}
where sequence of $\mathbf{x}^{(n)}$ was often saved to build Krylov subspace. 

> However, a great feature in his modified DMD acceleration algorithm is that a set successive differences $\mathbf{x}^{(n)}-\mathbf{x}^{(n-1)}$ is used as the snapshots in the data matrix $\mathbf{Y}_+$ and $\mathbf{Y}_-$ while the standard DMD with SVD decomposition is applied to form the approximation system $\mathbf{\tilde{A}}$.

Reviewer: ```suggestion
However, a great feature in his modified DMD acceleration algorithm is that a set successive differences $\mathbf{x}^{(n)}-\mathbf{x}^{(n-1)}$ is used as the snapshots in the data matrix $\mathbf{Y}_+$ and $\mathbf{Y}_-$, while the standard DMD with SVD decomposition is applied to form the approximation system $\mathbf{\tilde{A}}$.
```

The algorithm is as follows

> \begin{enumerate}

Reviewer: You may wish to make the algorithm a separate item using the algorithm2e package in latex.  This could then be referenced using \ALGORITHM{label}.

 \item Perform R source iterations: $\mathbf{x}^{l} = \mathbf{A} \mathbf{x}^{l-1} +b$
 \item Compute K source iterations to form $\mathbf{Y}_+$ and $\mathbf{Y}_-$. The last column of $Y_-$ we call $\mathbf{x}^{K-1}$ 
 \item Compute $\mathbf{x} = \mathbf{x}^{K-1} + \mathbf{U} \Delta y$.
\end{enumerate} 
where $U$ is the left unitary matrix from SVD decomposition of $\mathbf{Y}_-$ and $\Delta y$ is computed by 
\begin{equation}

>  (\mathbf{I} - \mathbf{\tilde{A}}) \Delta y = \mathbf{U}^T(\mathbf{x}^{K} - \mathbf{x}^{K-1})\, ,

Reviewer: ```suggestion
 (\mathbf{I} - \mathbf{\tilde{A}}) \Delta y = \mathbf{U}^T(\mathbf{x}^{K} - \mathbf{x}^{K-1})\, .
```

 \label{eq:McClarren}
\end{equation}

> Results from a homogeneous slab problem and a multi-dimensional pip problem suggest that a sequence of Richardson iterations followed by corrections reduces the number of iterations required by about one order of magnitude.

Reviewer: I need more description than this.  Citation? What slab?  What is a pip problem?

Author: Edited. This results is from McClarren's paper

> All of these past successes shows that DMD has potential to estimate the converged dominant modes of a reactor system by extrapolating forward from observed snapshots, even though the sequence of iterations from power method does not means ``real time''.

Reviewer: Add a paragraph to the end that says something like: 
In the following chapters the methods will be expanded in certain ways.

Author: I moved a paragraph about how the thesis organized to here.

\begin{equation}
 \mathbf{(I - DL^{-1}MS)} \mathbf{\mathbf{x}} = \frac{1}{k} \mathbf{DL^{-1}MF} \mathbf{\mathbf{x}}  \, ,
 \label{eq:keig}
\end{equation}
where $\mathbf{\mathbf{x}} $ describes the scalar flux, $\mathbf{L}$ represents loss operator,  $\mathbf{F}$ represents fission operator, $\mathbf{S}$ represents the scattering operator, $\mathbf{M}$ and $\mathbf{D}$ represent moment-to-discrete and discrete-to-moment operators respectively, and the eigenvalue $k$ represents the ratio of gains to losses.
The so called criticality of a reactor is determined by examining the dominant eigenvalue $k$, where $k=1$ is ``critical'' (the neutron population is steady), $k<1$ is ``subcritical'' (the neutron density is decreasing in time), and $k > 1$ is ``supercritical'' (the neutron density is increasing with time).
The eigenvector corresponding to the largest eigenvalue $k$, which is usually known as fundamental (or dominant) eigenmode, indicates the scalar flux distribution when the system reaches steady state.
A more generalized multigroup neutron transport equation will be presented in Chapter 2.

\section{Motivation and Objective}

> The discrete ordinates method, or the $S_n$ method, is a practical, deterministic approach that is used to discrize multigroup neutron transport equation.

Author: added

> This discrete ordinates equation is often solved by other iterative methods such as the power method (PM), the Richardson iteration method\cite{adams1993two}, or the generalized minimal residual method (GMRES)\cite{saad1986gmres} for a given level of accuracy. 

Reviewer: Discrete ordinates is a way of treating the angular and spatial variables in the transport problem. It isn't an equation to solve, but rather a way to approach solving the neutron transport equation.

Author: Lewis call it discrete ordinates equation, and said use iteration method to solve the  discrete ordinates equation, i think it is fine description of the process that discrete the flux and update the source.

> This discrete ordinates equation is often solved by other iterative methods such as the power method (PM), the Richardson iteration method\cite{adams1993two}, or the generalized minimal residual method (GMRES)\cite{saad1986gmres} for a given level of accuracy. 

Reviewer: ```suggestion
This discrete ordinates method is often implemented using an iterative method such as the power method (PM), the Richardson iteration method\cite{adams1993two}, or the generalized minimal residual method (GMRES)\cite{saad1986gmres} to solve for the dominant eigenpair. 
```

> No matter which iterative method is employed, it must apply the loss operator $L$ every iteration.

Reviewer: ```suggestion
While the specifics for of implementation differ for each iterative methods, Each of them begin with an initial approximation for the dominant eigenmode.
This approximation is inserted into the transport equation, and is used to solve for an improved approximation, by applying the loss operator $L$.
This process continues until the eigenmode has converged to some tolerance.
```

> Numerically, it is inefficient to solve for the operator inverse; instead, a series of sweeps over each energy group and space location can be performed until the system converges.  

Reviewer: ```suggestion
While possible in some cases to solve for the inverse of $L$, it is inefficient or impossible for large problems.
The various iterative methods provide a way of solving the operator equation without forming the matrix inverse by applying $L$ successively, which involves sweeping over each spatial cell and energy group.  
```

> However, the expense of this approach is now proportional to the number of iterations.

Reviewer: ```suggestion
The expense of these iterative methods is now proportional to the number of iterations, i.e., applications of the operator.
```

> Many breakthrough, such as the generalized Davidson(GD) method\cite{larsen1984diffusion} and Diffusion Synthetic Acceleration(DSA)\cite{hamilton2011numerical}, came to accelerate the aforementioned iteration methods.

Reviewer: ```suggestion
Methods such as the generalized Davidson(GD) method\cite{larsen1984diffusion} and diffusion synthetic acceleration(DSA)\cite{hamilton2011numerical} are designed to to accelerate the iteration methods by incorporating a correction to the approximation at each iteration.
```

Within the development of computational science，some theory of data science has been employed to analyze engineering problems.
Dynamic mode decomposition(DMD) is one of the most well-known, supervised learning approaches that has achieved great success in computational fluid dynamics.
This data-driven technology can produce reduced-dimensional surrogate models by gathering modes from given states of time dependent data and reconstructing the output parameters.
The primary focus of this thesis is to estimate accurate fundamental eigenmodes by DMD to accelerate the Power Method and other, related methods.

> Before continuing, it would be useful to briefly introduce some great achievements that demonstrated the success of dynamic mode decomposition.

Reviewer: ```suggestion
```

Author: I moved this sentence to the beginning of next section

\section{Summary of previous work}

DMD was initially proposed by \citet{schmid_dynamic_2010}\cite{schmid_applications_2011} to extract dynamics information from time-series data of fluids observations in 2010.

> He applied this frame on both numerical Navier-Stokes code results and experimental measures and illustrated the significant essential on quantifying fluid-dynamical problems.

Reviewer: `essential` is not the right word.  It doesn't make sense.

Author: This should be potential.

Author: maybe ''effects'' is better.

> As opposed to proper orthogonal decomposition (POD)\cite{lumley2007stochastic}, DMD is a pure data-based procedure and does not decompose on an orthogonal space. 

Reviewer: ```suggestion
As opposed to proper orthogonal decomposition (POD)\cite{lumley2007stochastic}, DMD is a purely data-based procedure and does not decompose on an orthogonal space. 
```

This method is based on Koopman analysis\cite{lasota2013chaos,mezic2005spectral} of nonlinear dynamical systems and can extract dynamic modes to describe the global behavior captured by projecting large-scale problems onto a reduced order dynamical system.
In his work, the time-dependent dynamic systems are described in the form of a snapshots sequence,
\begin{equation}
 \mathbf{X}^{N}_1 = \{\mathbf{x}_0, \mathbf{x}_1, \ldots, \mathbf{x}_{N} \} \, ,
 \label{eq:snap_matrix}
\end{equation}
where $\mathbf{x}_i$ stands for the {\it i}th state.
We assume that there exist a linear mapping operator $\mathbf{A}$, which can be applied on on the initial data snapshot $\mathbf{x}_0$ repeatedly and formulate snapshots sequence \EQ{eq:snap_matrix} as a Krylov subspace by
\begin{equation}
 \mathbf{X}^{N}_1 = \{\mathbf{x}_1,A\mathbf{x}_1,A^2\mathbf{x}_1,…,A^{N-1}\mathbf{x}_1 \} \, .
 \label{eq:Krylov_seq}
\end{equation}

> Some famous older attempts, like Arnoldi method\cite{arnoldi1951principle}, uses QR-decomposition $\mathbf{QR} = \mathbf{X}^{N-1}_1$ to reconstruct the mapping operator $\mathbf{A}$, which is not applicable because the explicit matrix form of  $\mathbf{A}$ is not often available for the projections.\cite{Greenbaum_1997, trefethen1997numerical}

Reviewer: ```suggestion
Some famous older attempts, such as the Arnoldi method\cite{arnoldi1951principle}, use QR-decomposition $\mathbf{QR} = \mathbf{X}^{N-1}_1$ to reconstruct the mapping operator $\mathbf{A}$, which is not possible if  the explicit matrix form of  $\mathbf{A}$ is unavailable \cite{Greenbaum_1997, trefethen1997numerical}.
When solving the neutron transport equation, the mapping operator is often unavailable due to the size of the system and the cost associated with explicitly forming this matrix.
```

Instead of employing the QR-decomposition, Schmid took advantage of the SVD decomposition and obtained a robust approximation by 
\begin{equation}
\mathbf{\tilde{A}} = \mathbf{U^H X_1^{N-1}V\Sigma^{-1}} \, ,
 \label{eq:stanard_DMD}
\end{equation}
where $\mathbf{X}_1^{N-1} = \mathbf{U}\Sigma \mathbf{V}^H$. 
The eigenvector $x_i$ of $\mathbf{\tilde{A}}$ and left eigenvector matrix $\mathbf{U}$ are used to calculate the DMD modes $\mathbf{\mathbf{x}}_i$ by 
\begin{equation}
\mathbf{\mathbf{x}}_i = \mathbf{U} \mathbf{x}_i \, ,
 \label{eq:dyanmic_modes}
\end{equation}
and, therefore, recover the reduced order dynamic system.
This scheme is often known as the standard DMD approach, and a more detailed discussion of the standard DMD scheme is presented in \CHAPTER{chapter:DMD-PM}.   

> Many practical theories were developed based on the this successfully attempt.

Reviewer: ```suggestion
Many practical theories were developed based on this successfully attempt.
```

One such attempt uses the past and future snapshot matrix, which is defined as $\mathbf{X}_0 \triangleq \{\mathbf{x}_0, \mathbf{x}_1, \ldots, \mathbf{x}_{N-1} \}$ and $\mathbf{X}_1 \triangleq \{\mathbf{x}_1, \mathbf{x}_1, \ldots, \mathbf{x}_{N} \} $. 
\citet{tu_dynamic_2014} posted a delicate mathematical analysis of the DMD algorithm and developed a theoretical extension of DMD (extract DMD) using a new definition of operater $\mathbf{A}$ by    
\begin{equation}
 \mathbf{A}\triangleq \mathbf{X}_1  \mathbf{X}_0^{\dagger}\, ,
 \label{eq:exact_dmd}
\end{equation}
where $\mathbf{X}_0^{\dagger}$ is the pseudoinverse of $\mathbf{X}_0$. 

> And the DMD modes and eigenvalues can be computed by eigendecomposition, however, it may be expensive to apply the eigendecomposition of $\mathbf{A}$in practice.

Reviewer: ```suggestion
The DMD modes and eigenvalues can be computed by eigendecomposition, however, it may be expensive to apply the eigendecomposition of $\mathbf{A}$ in practice.
```

> With very minimum changes, exact DMD can also compute the modes and eigenvalue without explicitly forming $\mathbf{A}$.

Reviewer: ```suggestion
With minimal changes, exact DMD can also compute the modes and eigenvalue without explicitly forming $\mathbf{A}$.
```

> In the case of directly-manipulation-free extract DMD, all processes are consistent with standard DMD, except for computation of the DMD modes, which is now done by

Reviewer: Are you sure the name is `directly-manipulation-free`? That doesn't sound right.

Author: edited

\begin{equation}
 \mathbf{\mathbf{x}}_i = \frac{1}{\lambda} \mathbf{X}_1 \mathbf{V} \mathbf{\Sigma}^{-1} \omega \, .
 \label{eq:exact_dmd_free}
\end{equation}

> \citet{kutz_dynamic_2016} summarized many other fashion variety DMD algorithms and discussed the applicability fitting with different complex systems in a recent monograph.

Reviewer: ```suggestion
\citet{kutz_dynamic_2016} summarized many variations on the DMD algorithm and discussed the applicability of each by solving with various complex systems in a recent monograph.
```

Fundamental theoretical foundations of DMD and the Koopman operator was also developed in this book.

> \subsection{DMD in Nuclear Engineering}

Reviewer: You do not need a subsection for 2 sentences...

Author: removed

Soon after, this technology then was applied to nuclear reactor simulations. 
\citet{abdo_data-driven_2018} used DMD as a direct, explicit-in-time surrogate for black-box models, e.g., to model the evolution of nuclear reactor isotopics over long time periods as well as the nonlinear response of reactor power during short transients.\cite{abdo_modeling_2019}\cite{elzohery2018comparison}

\subsection{DMD Accelerated Iterative Methods}

> A way to achieve acceleration is to use results from lower dimensional system projected system to find a correction each iteration until system converges.

Reviewer: ```suggestion
As mentioned briefly, a way to achieve acceleration of iterative methods is to correct the eigenmode at each iteration to reduce the total number of iterations.
One way to do this is to use the results from a lower-dimensional, projected system, to predict the solution to the higher-dimensional system that we wish solved.
```

> The first research of using DMD to accelerate iterative methods was the work of \citet{andersson_novel}.

Reviewer: ```suggestion
\citet{andersson_novel} first used DMD to accelerate iterative methods.
```

A Krylov subspace storing the state-samples with time sequence as $\mathbf{V_n}$ is built up in a same way as \EQ{eq:Krylov_seq}.
Then the Gram-Schmidt algorithm is applied by $\mathbf{V_n} = \mathbf{{E_n U_n}}$ to orthogonalize the subspace.
Most applications avoid the SVD because the Gram-Schmidt algorithm only works with two state vectors each time, and the SVD requires the whole subspace matrix $V_n$. 
Moreover, Gram-Schmidt algorithm is obviously a more suitable approach for parallel computations. 
The reduced projection of the system matrix $\mathbf{\tilde{A}}$ is obtained by 
\begin{equation}
 \mathbf{\tilde{A}} = \mathbf{E}_n^T \mathbf{V}_{n+1} \mathbf{U}_n^{-1}  \, ,
 \label{eq:andersson_reduce}
\end{equation}
and is implemented into iterative linear relation $\mathbf{x}^{(n+1)} = \mathbf{A}\mathbf{x}^{(n)} + \mathbf{b}$.

> The estimate of the converged solution can be calculate in the modified form as:

Reviewer: ```suggestion
The estimate of the converged solution can be calculated in the modified form as
```

\begin{equation}
 \mathbf{x}^{u} = \mathbf{x}^{(n+1)} + \mathbf{E}_n(\mathbf{I_n} - \mathbf{\tilde{A}}_n)^{-1} \mathbf{E}_n^T(\mathbf{x}^{(n+2)} - \mathbf{x}^{(n+1)}) \, .
 \label{eq:andersson}
\end{equation}
By performing this process several times to correct the solution, both the number of iterations and the average fluctuation were reduced almost 30\% for compressible flow problems. 

Then, \citet{mcclarren_acceleration_2018} presented a novel acceleration technique for improving the convergence of Richardson(source) iteration for source-driven neutronics problems.

> The Richardson iteration can be expressed as 

Reviewer: ```suggestion
Richardson iteration can be expressed as 
```

\begin{equation}
 \mathbf{x}^{(n+1)} = (\mathbf{I}-\mathbf{A})\mathbf{x}^{(n)} + \mathbf{b} \, ,
 \label{eq:richardson}
\end{equation}

> where sequence of $\mathbf{x}^{(n)}$ was often saved to build Krylov subspace. 

Reviewer: ```suggestion
where the sequence of $\mathbf{x}^{(n)}$ was often saved to build a Krylov subspace. 
```

> However, a great feature in his modified DMD acceleration algorithm is that a set successive differences $\mathbf{x}^{(n)}-\mathbf{x}^{(n-1)}$ is used as the snapshots in the data matrix $\mathbf{Y}_+$ and $\mathbf{Y}_-$, while the standard DMD with SVD decomposition is applied to form the approximation system $\mathbf{\tilde{A}}$.

Reviewer: ```suggestion
A feature in his modified DMD acceleration algorithm was that a set successive differences $\mathbf{x}^{(n)}-\mathbf{x}^{(n-1)}$ was used as the snapshots in the data matrix $\mathbf{Y}_+$ and $\mathbf{Y}_-$, while the standard DMD with SVD decomposition was applied to form the approximation system $\mathbf{\tilde{A}}$.
```

% | LaTeX will automatically number the chapters.
% +--------------------------------------------------------------------+

\chapter{Introduction}
\label{chapter:intro}

> Nuclear power was studied initially in 1940s and has become an important supply in power industry.

Reviewer: ```suggestion
Nuclear power was initially studied in the 1940s and has become an important source in the power industry.
```

> Without estimating the desired state of neutron population, it is impossible to control the neutronic systems and generate essential design. 

Reviewer: This line does not flow well with the previous

Author: fixed at 3802c04b06ac869a9db79cfd24b3c4716c8873f0

> Computational simulation technology is critical in nuclear physics, which does not only used to optimize nuclear reactor design, but also help to understanding extremely complicated neutron dynamic behaviors.

Reviewer: ```suggestion
In nuclear physics, computational simulations are critical for optimizing nuclear reactor designs and helping to understand extremely complicated neutron dynamic behaviors.
```

> However, to model a full core is an long-standing goal while it is still too computationally expensive to directly solve it even using the most advanced large-scale super computers. 

Reviewer: ```suggestion
However, modeling a full core is still too computationally expensive even with the most advanced, large-scale, super computers. 
```

> Therefore, having an efficient approach to obtain the criticality and neutron distribution accurately is a pivotal topic in this field.

Reviewer: ```suggestion
Therefore, having an accurate and efficient approach to determine the criticality and the neutron distribution is a pivotal topic in this field.
```

\section{Operator Notation}
The generalized eigenvalue form of the neutron transport equation is a fundamental problem in computational reactor physics.
The dominant eigenvector and corresponding eigenvalue of this equation can represent if a system is in steady-state conditions.

> Though criticality calculation arised from the analysis of time-dependent processes, it is sufficient to reduced transport equation in steady-state. 

Reviewer: ```suggestion
Though criticality calculations arose from the analysis of time-dependent processes, it is sufficient to reduce the transport equation to steady-state in many cases. 
```

> Then the transport equation can be written in operator notation as

Reviewer: ```suggestion
The steady-state transport equation can be written in operator notation as
```

\begin{equation}
 \mathbf{(I - DL^{-1}MS)} \mathbf{\phi} = \frac{1}{k} \mathbf{DL^{-1}MF} \mathbf{\phi}  \, ,
 \label{eq:keig}
\end{equation}
where $\mathbf{\phi} $ describes the scalar flux, $\mathbf{L}$ represents loss operator,  $\mathbf{F}$ represents fission operator, $\mathbf{S}$ represents the scattering operator, $\mathbf{M}$ and $\mathbf{D}$ represent moment-to-discrete and discrete-to-moment operators respectively, and the eigenvalue $k$ represents the ratio of gains to losses.

> A reactor is "critical," i.e., the neutron population is steady, when $k = 1$; or "subcritical," i.e., the neutron density is decreasing as time passes when $k < 1$; or "supercritical," i.e., the neutron density is increasing with time when $k>1$.

Reviewer: ```suggestion
A reactor is ``critical,'' i.e., the neutron population is steady, when $k = 1$; or ``subcritical,'' i.e., the neutron density is decreasing as time passes when $k < 1$; or ``supercritical,'' i.e., the neutron density is increasing with time when $k>1$.
```

> The eigenvector corresponding to the largest eigenvalue $k$, which is usually known as fundamental (or dominant) eigenmode, indicates the scalar flux distribution when the system reach steady state.

Reviewer: ```suggestion
The eigenvector corresponding to the largest eigenvalue $k$, which is usually known as fundamental (or dominant) eigenmode, indicates the scalar flux distribution when the system reaches steady state.
```

> A more generalized multigroup neutron transport equation will be represented in Chapter 2.

Reviewer: ```suggestion
A more generalized multigroup neutron transport equation will be presented in Chapter 2.
```

\section{Motivation and Objective}

> Discrete ordinates method is a practical deterministic approach that used to solve Equation \ref{eq:keig}.

Reviewer: ```suggestion
The discrete ordinates method is a practical, deterministic approach that is used to solve Equation \ref{eq:keig}.
```

Reviewer: In main.tex, you have several macros setup to cite equations and figures.  You should use them!

> This approach is often smoothed by other iterative method, like power method, Richard iteration method and GMRES, and will apply the loss operator $L$ every iteration.

Reviewer: ```suggestion
This approach is often smoothed by other iterative methods such as the power method, Richard iteration method, or GMRES, which each apply the loss operator $L$ every iteration.
```

Reviewer: I'm not sure what you mean by smoothed in that sentence

> Numerically, we don't solve directly inverse the operator, and would sweeping over each energy group and space location until the system converges instead.  

Reviewer: ```suggestion
Numerically, it is inefficient to solve for the operator inverse; instead, a series of sweeps over each energy group and space location can be performed until the system converges.  
```

> However, it will still cost the enormous computational resource when it requires a large amount number of iterations. 

Reviewer: ```suggestion
However, the expense of this approach is now proportional to the number of iterations.
```

> Many breakthrough, such as the Generalized Davidson(GD) method and Diffusion Synthetic Acceleration(DSA), came to accelerate those iteration methods.

Reviewer: ```suggestion
Many breakthrough, such as the Generalized Davidson(GD) method and Diffusion Synthetic Acceleration(DSA), came to accelerate the aforementioned iteration methods.
```

Reviewer: You should have citations for all of these methods

> Within the development of computational science，some theory of data science has been employed to analyze engineering problems.

Reviewer: ```suggestion
```

> Dynamic mode decomposition(DMD) was one of the most well-known supervised learning approach that has achieved great success in computational fluid dynamics.

Reviewer: ```suggestion
Dynamic mode decomposition(DMD) is one of the most well-known supervised learning approaches that has achieved great success in computational fluid dynamics.
```

> This data-driven technology could produce reduced-dimensional surrogate model by gathering modes from given states of the time series and reconstructing the output parameters.

Reviewer: ```suggestion
This data-driven technology can produce reduced-dimensional surrogate models by gathering modes from given states of time dependent data and reconstructing the output parameters.
```

> The primary focus of this dissertation is to estimate a more accurate fundamental eigenmodes by DMD to accelerate power and related methods.

Reviewer: ```suggestion
The primary focus of this thesis is to estimate accurate fundamental eigenmodes by DMD to accelerate the Power Method and other, related methods.
```

> To achieve these goals, a reviews of applications of DMD will be described in the rest of Chapter 1.

Reviewer: ```suggestion
To achieve these goals, a review of DMD will be described in the remainder of this chapter.
```

> Chapter 2 will introduce the multigroup neutron transport and diffusion equation, and Chapter 3 discuss the mathematically background of power and flatten power method.

Reviewer: ```suggestion
Chapter 2 will introduce the multigroup neutron transport and diffusion equations, and Chapter 3 discusses the mathematical background of the power and flatten power methods.
```

> Following that, DMD-PM(n) are presented in Chapter 4, while DMD-FPM(n) are presented in Chapter 5.

Reviewer: You have to define acronyms like DMD-PM(n) before using them.

Author: I define both of acronyms at abstract.

Reviewer: The abstract does not count as part of the document.  You must define them both in the abstract and the main document.

Author: fixed at 19a7a6f367a9a4a41327cb0161b1e1121fe6b4da

Then, the results and further analysis of the 1-D and 2-D test problems are described in Chapter 6.

> Chapter 7 will include conclusions and future work including possibility to combine this theory with other methods. 

Reviewer: ```suggestion
Chapter 7 will include conclusions and future work including the possibility of combining this theory with other methods. 
```

\section{Summary of previous work}

> Before continuing, it would be useful to brief introduce some great achievements that demonstrated the success of dynamic mode decompositio

Reviewer: ```suggestion
Before continuing, it would be useful to briefly introduce some great achievements that demonstrated the success of dynamic mode decomposition.
```

DMD was initially proposed by Schmid to extract dynamics information from time-series data of fluids observations.\cite{schmid_dynamic_2010}\cite{schmid_applications_2011}
Schmid proposed a decomposition technology, dynamic mode decomposition(DMD), based on Koopman analysis of nonlinear dynamical systems. 

> This method extracted dynamic modes describe the global behavior captured from numerical and experimental data to project large-scale problems onto a reduced order dynamical system. 

Reviewer: ```suggestion
This method extracted dynamic modes to describe the global behavior captured from numerical and experimental data by projecting large-scale problems onto a reduced order dynamical system. 
```

> Compare to proper orthogonal decomposition(POD), DMD directly applied on data and does not decomposite on orthogonal space. 

Reviewer: ```suggestion
As opposed to proper orthogonal decomposition(POD), DMD is applied directly to the data and does not decomposite on an orthogonal space. 
```

> Schmid applied this frame on both numerical Navier-Stokes code results and experimental measures and illustrated the significant essential on quantifying fluid-dynamical problems.

Reviewer: This doesn't make sense

> Schmid shows the initial idea of DMD algorithm which is usually known as standard DMD algorithm in the subsequent publications. 

Reviewer: ```suggestion
Schmid shows the initial idea of the DMD algorithm, which is called the standard DMD algorithm in the remainder of this work. 
```

> Instead of employing QR-decomposition, he took advantage of SVD decomposition and obtain a robust approximation by $\tilde{S} = U^HV_2^NW\Sigma^{-1}$, where $V^{N-1}_1 = U\Sigma W^H$. 

Reviewer: ```suggestion
Instead of employing the QR-decomposition, he took advantage of the SVD decomposition and obtained a robust approximation by $\tilde{S} = U^HV_2^NW\Sigma^{-1}$, where $V^{N-1}_1 = U\Sigma W^H$. 
```

Reviewer: Shouldn't the matrices be bolded?

> The eigenvector $y_i$ of matrix $S$ and left eigenvector matrix U would be used to form the final dynamic modes $\phi_i$. Compare to the standard DMD, the tradition method use the least square solution to form matrix $S$ from QR-decomposition by $a = R^{−1}Q^Hv_N$, where $QR = V^{N-1}_1$. 

Reviewer: ```suggestion
The eigenvector $y_i$ of matrix $S$ and left eigenvector matrix U would be used to form the final dynamic modes $\phi_i$.
Compared to the standard DMD, the traditional method uses the least squared solution to form matrix $S$ from QR-decomposition by $a = R^{−1}Q^Hv_N$, where $QR = V^{N-1}_1$. 
```

> Also, some older attempts using Arnoldi method and Hessenberg matrix by $AQ \approx QH$ and $H = RSR^{−1}$ is not applicable because projections the matrix $A$ has to be available. 

Reviewer: ```suggestion
Also, some older attempts using the Arnoldi method and the Hessenberg matrix by $AQ \approx QH$ and $H = RSR^{−1}$ is not applicable because projections the matrix $A$ has to be available. 
```

Here $V^{N}_1 = \{v_1,Av_1,A^2v_1,…,A^{N-1}v_1 \} $ is the Krylov sequence based on the initial data snapshot $v_1$ and the linear map $A$ connect two snapshots ordered by sequence.  

> Tu and Kutz posted mathematical analysis of DMD algorithm and further extension including Extract DMD.  \cite{tu_dynamic_2014}

Reviewer: ```suggestion
\citep{tu_dynamic_2014} posted mathematical analysis of the DMD algorithm and further extension including Extract DMD.  
```

Reviewer: \citet{bib} will give the author names while \citep{bib} will just put the citation

> They also summarized many other fashion variety DMD algorithms, such as mrDMD and DMDc, and discuss the applicability fitting with different complex systems.\cite{kutz_dynamic_2016}

Reviewer: You have listed so many different variations of DMD and I want more details on the differences between them.

 \label{eq:richardson}
\end{equation}
where sequence of $\mathbf{x}^{(n)}$ was often saved to build Krylov subspace. 
However, a great feature in his modified DMD acceleration algorithm is that a set successive differences $\mathbf{x}^{(n)}-\mathbf{x}^{(n-1)}$ is used as the columns in the data matrix $Y_+$ and $Y_-$ while the standard DMD with SVD decomposition is applied to form the approximation system $\tilde{A}$.
The algorithm is as follows
\begin{enumerate}
 \item Perform R source iterations: $\phi^{l} = A \phi^{l-1} +b$
 \item Compute K source iterations to form $Y_+$ and $Y_-$. The last column of $Y_-$ we call $\phi^{K-1}$ 
 \item Compute $\phi = \phi^{K-1} + U \Delta y$.
\end{enumerate} 
Here, $U$ is the left unitary matrix from SVD decomposition of $Y_-$ and $\Delta y$ is computed by 
\begin{equation}
 (I - \tilde{A}) \Delta y = U^T(\phi^{K} - \phi^{K-1})\, ,
 \label{eq:McClarren}
\end{equation}
Results from a homogeneous slab problem and a multi-dimensional pip problem suggest that a sequence of Richardson iterations followed by corrections reduces the number of iterations required by about one order of magnitude.

> All of these past successes shows that DMD has potential to estimate the converged dominant modes of a reactor system by extrapolating forward from observed snapshots, even though the sequence of iterations from power method does not means "real time".

Reviewer: If you use `"` then the leading marks will be backwards in the final document.  You should use ` `` ` to begin the quote and `''` to end.

> All of these past successes shows that DMD has potential to estimate the converged dominant modes of a reactor system by extrapolating forward from observed snapshots, even though the sequence of iterations from power method does not means "real time".

Reviewer: ```suggestion
All of these past successes shows that DMD has potential to estimate the converged dominant modes of a reactor system by extrapolating forward from observed snapshots, even though the sequence of iterations from power method does not means ``real time''.
```
